src/praisonai/praisonai/sandbox/sandlock.py

```
@@ -0,0 +1,471 @@
+        self._temp_dir: Optional[str] = None
+        self._sandbox = None
+        
+        # Lazy import sandlock to avoid import-time dependency
+        try:
+            import sandlock
+            self._sandlock = sandlock
+        except ImportError:
+            raise ImportError(
+                "sandlock package required for SandlockSandbox. "
+                "Install with: pip install 'praisonai[sandbox]' or pip install sandlock"
+            )
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)